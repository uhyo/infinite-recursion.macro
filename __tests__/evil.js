const pluginTester = require("babel-plugin-tester").default;
const plugin = require("babel-plugin-macros");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: {
    globalName: `
      import rec from '../../infinite-recursion.macro'
      const _runRecursive = 3;

      console.log(rec(function sumTo(num){
        return num <= 0 ? 0 : num + sumTo(num - 1);
      }, 1e7));
    `,
    local: `
      import rec from '../../infinite-recursion.macro'

      console.log(rec(function sumTo(num){
        const _runRecursive = 3;
        return num <= 0 ? 0 : num + sumTo(num - 1);
      }, 1e7));
    `,
  },
});
