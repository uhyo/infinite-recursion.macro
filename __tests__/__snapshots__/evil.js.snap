// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros globalName: globalName 1`] = `

import rec from '../../infinite-recursion.macro'
const _runRecursive = 3;

console.log(rec(function sumTo(num){
  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

import { runRecursive as _runRecursive2 } from "../../infinite-recursion.macro/lib/runtime.js";
const _runRecursive = 3;
console.log(
  _runRecursive2(function* sumTo(num) {
    return num <= 0 ? 0 : num + (yield [num - 1]);
  }, 1e7)
);


`;

exports[`macros local: local 1`] = `

import rec from '../../infinite-recursion.macro'

console.log(rec(function sumTo(num){
  const _runRecursive = 3;
  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

import { runRecursive as _runRecursive2 } from "../../infinite-recursion.macro/lib/runtime.js";
console.log(
  _runRecursive2(function* sumTo(num) {
    const _runRecursive = 3;
    return num <= 0 ? 0 : num + (yield [num - 1]);
  }, 1e7)
);


`;
